artpaint/layers/LayerWindow.cpp

```diff
@@ -46,28 +63,87 @@ sem_id LayerWindow::layer_window_semaphore = create_sem(1,"layer window semaphor
 LayerWindow::LayerWindow(BRect frame)
 	: BWindow(frame, B_TRANSLATE("Layers"),
 		B_FLOATING_WINDOW_LOOK, B_NORMAL_WINDOW_FEEL,
-		B_NOT_H_RESIZABLE | B_NOT_ZOOMABLE )
+		B_NOT_ZOOMABLE)
 {
-	BBox *top_part = new BBox(BRect(-1, 0, Bounds().Width() + 2,
-		HS_MINIATURE_IMAGE_HEIGHT + 3), NULL, B_FOLLOW_LEFT_RIGHT | B_FOLLOW_TOP);
-	AddChild(top_part);
-	title_view = new BStringView(BRect(HS_MINIATURE_IMAGE_WIDTH + 10, 2,
-		top_part->Bounds().Width() - 2, HS_MINIATURE_IMAGE_HEIGHT - 2),
-		"image title", "", B_FOLLOW_LEFT_RIGHT | B_FOLLOW_TOP);
-	top_part->AddChild(title_view);
+	top_part = new BBox(B_PLAIN_BORDER, NULL);
+	title_view = new BStringView("title", "");
+
+	layer_operation_menu = new BMenu(B_TRANSLATE("Actions"));
+
+	layer_operation_menu->AddItem(new BMenuItem(
+		B_TRANSLATE("Merge with layer above"),
+		new BMessage(HS_MERGE_WITH_UPPER_LAYER)));
+
+	layer_operation_menu->AddItem(new BMenuItem(
+		B_TRANSLATE("Merge with layer below"),
+		new BMessage(HS_MERGE_WITH_LOWER_LAYER)));
+
+	layer_operation_menu->AddSeparatorItem();
+
+	layer_operation_menu->AddItem(new BMenuItem(
+		B_TRANSLATE("Add layer above"),
+		new BMessage(HS_ADD_LAYER_FRONT)));
+
+	layer_operation_menu->AddItem(new BMenuItem(
+		B_TRANSLATE("Add layer below"),
+		new BMessage(HS_ADD_LAYER_BEHIND)));
+
+	layer_operation_menu->AddSeparatorItem();
+
+	layer_operation_menu->AddItem(new BMenuItem(
+		B_TRANSLATE("Duplicate layer"),
+		new BMessage(HS_DUPLICATE_LAYER)));
+
+	layer_operation_menu->AddSeparatorItem();
+
+	layer_operation_menu->AddItem(new BMenuItem(
+		B_TRANSLATE("Delete layer"),
+		new BMessage(HS_DELETE_LAYER)));
+
+	layer_operation_menu->SetRadioMode(false);
+
+	BMenuBar* menu = new BMenuBar("menu bar");
+
+	menu->AddItem(layer_operation_menu);
+
+	BMessage* message = new BMessage(HS_LAYER_TRANSPARENCY_CHANGED);
+	message->AddInt32("value", 0);
+
+	transparency_slider =
+		new NumberSliderControl(B_TRANSLATE("\xCE\xB1:"), "0",
+		message, 0, 100, false);
+
+	BFont font;
+
+	BGridLayout* transparencyLayout = BLayoutBuilder::Grid<>(0.0)
+		.Add(transparency_slider, 0, 0, 0, 0)
+		.Add(transparency_slider->LabelLayoutItem(), 0, 0)
+		.Add(transparency_slider->TextViewLayoutItem(), 1, 0)
+		.Add(transparency_slider->Slider(), 2, 0, 2);
+	transparencyLayout->SetMaxColumnWidth(1, font.StringWidth("1"));
+	transparencyLayout->SetMinColumnWidth(2,
+		font.StringWidth("SLIDERSLIDERSLIDER"));
+
+	transparency_slider->Slider()->SetToolTip(B_TRANSLATE("Set Layer Transparency"));
```

Comment:
Sentence casing: "Set layer transparency" or maybe just "Layer transparency" as the whole widget also just shows the layer transparency, if the user doesn't twiddle the slider knob.